server/plugin.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"sync"
+
+	"github.com/mattermost/mattermost-server/mlog"
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/plugin"
+	"github.com/pkg/errors"
+	"github.com/robfig/cron"
+)
+
+// Plugin implements the interface expected by the Mattermost server to communicate between the server and plugin processes.
+type Plugin struct {
+	plugin.MattermostPlugin
+
+	// configurationLock synchronizes access to the configuration.
+	configurationLock sync.RWMutex
+
+	// configuration is the active plugin configuration. Consult getConfiguration and
+	// setConfiguration for usage.
+	configuration *configuration
+
+	// a job for pre-calculating channel recommendations for users.
+	preCalcJob    *cron.Cron
+	preCalcPeriod string
+	botUserID     string
+}
+
+type readFile func(filename string) ([]byte, error)
+
+func (p *Plugin) setupBot(reader readFile) error {
+	botID, err := p.Helpers.EnsureBot(&model.Bot{
+		Username:    "suggestions",
+		DisplayName: "Suggestions",
+		Description: "Created by the Suggestions plugin.",
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to ensure suggestions bot")
+	}
+	p.botUserID = botID
+	bundlePath, err := p.API.GetBundlePath()
+	if err != nil {
+		return errors.Wrap(err, "couldn't get bundle path")
+	}
+
+	profileImage, err := reader(filepath.Join(bundlePath, "assets", "profile.jpeg"))
+	if err != nil {
+		return errors.Wrap(err, "couldn't read profile image")
+	}
+
+	appErr := p.API.SetProfileImage(botID, profileImage)
+	if appErr != nil {
+		return errors.Wrap(appErr, "couldn't set profile image")
+	}
+	return nil
+}
+
+func (p *Plugin) startPrecalcJob() error {
+	config := p.getConfiguration()
+	p.preCalcPeriod = "@weekly" // Run once a week, midnight between Sat/Sun
```

Comment:
1/5 you should use the default value instead: https://developers.mattermost.com/extend/plugins/manifest-reference/#settings_schema.settings.default